### PR TITLE
Use makefile for terraform

### PR DESCRIPTION
**Pull request by chelnak**
## Context

This PR implements https://trello.com/c/oio959xP/442-apply-use-makefile-to-run-terraform-in-deploy-workflow

## Changes proposed in this pull request

* (roughly) Align deploy steps with public
* Ensure deploy.yml uses Makefile for infra deploys

## Link to Trello card

[](https://trello.com/c/oio959xP/442-apply-use-makefile-to-run-terraform-in-deploy-workflow)

## Things to check

- [x] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -130,10 +130,7 @@ jobs:
           CONFIRM_PRODUCTION: true
 
       - name: Terraform init, plan & apply
-        run: |
-          terraform init -backend-config=workspace_variables/${{ matrix.environment }}_backend.tfvars
-          terraform plan -var-file workspace_variables/${{ matrix.environment }}.tfvars.json -out tfplan
-          terraform apply -auto-approve -input=false "tfplan"
+        run: make ${{ matrix.environment }} ci deploy
```
Comment (saliceti) [inline .github/workflows/deploy.yml:133]: TF_VAR_paas_docker_image would be overwritten by "main" if IMAGE_TAG is not passed

Should we set the IMAGE_TAG env var to the commit SHA as per https://github.com/DFE-Digital/register-trainee-teachers/blob/main/.github/workflows/deploy.yml#L120 instead?

File: Makefile [hunk 2]
```diff
@@ -132,21 +136,23 @@ shell: ## Open a shell on the app instance on PaaS, eg: make qa shell
...
-	$(if $(PASSCODE), , $(error Missing environment variable "PASSCODE", retrieve from https://login.london.cloud.service.gov.uk/passcode))
+	$(if $(or $(DISABLE_PASSCODE),$(PASSCODE)), , $(error Missing environment variable "PASSCODE", retrieve from https://login.london.cloud.service.gov.uk/passcode))
 	$(eval export TF_VAR_paas_sso_code=$(PASSCODE))
-	az account set -s $(AZURE_SUBSCRIPTION) && az account show \
-	&& cd terraform && terraform init -reconfigure -backend-config=workspace_variables/$(APP_ENV)_backend.tfvars
+	$(eval export TF_VAR_paas_docker_image=ghcr.io/dfe-digital/apply-teacher-training:$(IMAGE_TAG))
+
+	az account set -s $(AZURE_SUBSCRIPTION) && az account show
+	cd terraform && terraform init -reconfigure -backend-config=workspace_variables/$(APP_ENV)_backend.tfvars
+	echo "🚀 APP_ENV is $(APP_ENV)"
```
Comment (saliceti) [inline Makefile:145]: This may not be necessary going forward

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (chelnak) [reply .github/workflows/deploy.yml:133]: I've updated the env vars to include IMAGE_TAG

File: Makefile (hunk 2 above)
Comment (chelnak) [reply Makefile]: Removed!

**Review by saliceti — APPROVED** [review]
👏

**Comment by saliceti** [comment]
[Apply: Use Makefile to run terraform in deploy workflow](https://trello.com/c/oio959xP/442-apply-use-makefile-to-run-terraform-in-deploy-workflow)

**Pull request merged**